docs/docs/core/strapi/entity-service.md

```diff
@@ -0,0 +1,33 @@
+---
+title: Entity Service
+tags:
+  - strapi
+  - service
+  - entity-service
+---
+
+# Entity Service
+Strapi provides an Entity Service API, built on top of the Query Engine API (entity-manager in the Database Layer). The Entity Service is the layer that handles Strapi's complex data structures like components and dynamic zones, and uses the Query Engine API under the hood to execute database queries.
+## Decorators
+The Entity Service uses the decorator pattern so plugins and Strapi core packages can insert decorators to change the output of the Entity Service function.
+
+Examples of decorators are plugins/i18n and the core/upload.  You can decorate it like:
+
+
+SOME CODE  EXAMPLE
+
+
+### decorators
+When writing a decorator, please make sure you follow this checklist:
+- You should call `service.(function).call()` in your decorator function. the consequence of not doing this is that you are not calling the decorators bellow this decorator. What would mean you ignore them.
+- You are recommend to add your changes to the output of `service.(function).call()` so you can return it with the output of the lower laying decorators.
+- It is best to use `opts`/`options` you get from the parent decorator into your service call into `service.(function).call()`. You can add stuff to it but please remember to not change the options you get. Since change them would have as an effect that it could break the decorator above it.
```

Comment:
```suggestion
- It is best to use the `opts`/`options` arguments received from the parent decorator in the `service.(function).call()` function. You can add to it but please remember to not change the received options. Changing the options could break the decorator above it.
```

Reply:
When you say you can add stuff to it, by "it" do you mean the options? or do you mean you can add more arguments to the call function?

Reply:
Will also spesify this better by it I mean to the opts or options variables but you should not change anyting in it already since that would break layers that have run before it.